apps/vaporgui/TrackBall.cpp

```diff
@@ -368,11 +370,26 @@ bool Trackball::ReconstructCamera(
 
 // Set the quaternion and translation from a viewer frame
 // Also happens to construct modelview matrix, but we don't use its translation
-void Trackball::setFromFrame(
+int Trackball::setFromFrame(
 	const std::vector<double>& posvec, const std::vector<double>& dirvec, 
 	const std::vector<double>& upvec, const std::vector<double>& centerRot,
 	bool persp
 ) {
+    VAssert( posvec.size() == 3 );
+    VAssert( dirvec.size() == 3 );
+    VAssert( upvec.size() == 3 );
+    VAssert( centerRot.size() == 3 );
+
+    // Check for valid up vector and view direction
+    if ( upvec[0] == 0 && upvec[1] == 0 && upvec[2] == 0 ) {
+        MSG_ERR("Invalid up vector");
```

Comment:
Leave the error reporting to the user of this class, or make use of MyBase to log the error if you think that it's important to provide a message.

Reply:
Ok, error report has been bumped up into the NavigationEventRouter.